autoconfigure/collector-sqs/src/main/java/zipkin/autoconfigure/collector/sqs/ZipkinSQSCollectorAutoConfiguration.java

```diff
@@ -37,8 +42,28 @@
   /** By default, get credentials from the {@link DefaultAWSCredentialsProviderChain */
   @Bean
   @ConditionalOnMissingBean
-  AWSCredentialsProvider credentialsProvider() {
-    return new DefaultAWSCredentialsProviderChain();
+  AWSCredentialsProvider credentialsProvider(ZipkinSQSCollectorProperties properties) {
+    // Create credentials provider from ID and secret if given otherwise use the default provider
+    AWSCredentialsProvider provider;
+    if (!isNullOrEmpty(properties.awsAccessKeyId) && !isNullOrEmpty(properties.awsSecretAccessKey)) {
+      provider = new AWSStaticCredentialsProvider(
+          new BasicAWSCredentials(properties.awsAccessKeyId, properties.awsSecretAccessKey));
+    } else {
+      provider = new DefaultAWSCredentialsProviderChain();
+    }
+
+    // Create STS provider if a role is specified
+    if (!isNullOrEmpty(properties.awsStsRoleArn)) {
```

Comment:
That makes sense.  Creating the STS service and credentials provider are a bit of a chicken and egg thing.  I took a stab at splitting this out and hopefully achieved what you had in mind.